### ISSUE TITLE
[BUG] GSConnect crashes on startup due to GLib.FileError - cannot create Google Chrome file that already exists

### ISSUE THREAD

**Issue by kogli** [andyholmes/gnome-shell-extension-gsconnect#770]
**Describe the bug**

Upon system startup, the GSConnect extension fails to load and does not work. The extensions.gnome.org website shows "ERROR" message.

The Alt+F2 `lg` command shows the following error message (maybe not the exact words because I translate it from Czech to English):
```
GLib.FileError: Could not create file "/home/nojmy/.config/google-chrome/NativeMessagingHosts/org.gnome.shell.extensions.gsconnect.json.XDAWG0...
```

If I go into the aforementioned directory, the file `org.gnome.shell.extensions.gsconnect.json` (without the appendix) exists. Despite that, the extension crashes and does not work.

Please note that all the directories in the file system hierarchy are owned by `nojmy:nojmy` (my username) and have read and write permissions for my user.

Opening extensions.gnome.org, removing and re-installing GSConnect results in GSConnect working normally for the rest of my session (all the configuration stays preserved if I do this).

I tried reinstalling Google Chrome and this actually worked for a while, but now this unexpected behavior is back for some reason. (I mostly use FIrefox, but I do use Google Chrome occasionally.)

**Steps To Reproduce:**

1. Have GSConnect and Google Chrome installed
2. Upon system startup, GSConnect doesn't start (the reason is unknown - probably that the aforementioned file already exists?)
3. Press Alt+F2
4. Type `lg` and press Enter
5. Open the "Extensions" tab
6. Find GSConnect in the extensions list and press "display errors"
7. See error message

**Expected behavior**

GSConnect sees that the file is already in place as it should and continues operating as usual.

**Support Log**

Generating a support log is pointless as the extension is not operational (and therefore I do not have access to the "generate support log" button) and upon reinstallation of the extension, the support log is empty.

**System Details (please complete the following information):**

 - **GSConnect version:** 31
   - **Installed from:** GNOME Extensions Websitete
 - **GNOME/Shell version:** 3.34.4
 - **Distro/Release:** Manjaro (newest rolling-release)

**Additional Notes:**

What I find odd is that `~/.config/google-chrome` is actually a symlink to `/run/user/1000/nojmy-google-chrome`, where `nojmy` is my username. This should however not be a problem due to the fact that all directory/file permissions are set correctly and are owned by the `nojmy:nojmy` user/group, as can be seen below:

```bash
[nojmy@Nojmy-XPS ~]$ ls -ld ~/.config/
drwxr-xr-x 1 nojmy nojmy 1900 Mar  3 12:14 /home/nojmy/.config/
[nojmy@Nojmy-XPS ~]$ ls -ld ~/.config/google-chrome
lrwxrwxrwx 1 nojmy nojmy 34 Mar  3 12:07 /home/nojmy/.config/google-chrome -> /run/user/1000/nojmy-google-chrome
[nojmy@Nojmy-XPS ~]$ ls -ld /run/user/1000/nojmy-google-chrome/
drwxr-xr-x 23 nojmy nojmy 580 Mar  3 12:07 /run/user/1000/nojmy-google-chrome/
[nojmy@Nojmy-XPS ~]$ ls -ld /run/user/1000/nojmy-google-chrome/NativeMessagingHosts/
drwxr-xr-x 2 nojmy nojmy 60 Mar  3 12:14 /run/user/1000/nojmy-google-chrome/NativeMessagingHosts/
[nojmy@Nojmy-XPS ~]$ ls -ld /run/user/1000/nojmy-google-chrome/NativeMessagingHosts/org.gnome.shell.extensions.gsconnect.json 
-rw-r--r-- 1 nojmy nojmy 355 Mar  3 12:14 /run/user/1000/nojmy-google-chrome/NativeMessagingHosts/org.gnome.shell.extensions.gsconnect.json
[nojmy@Nojmy-XPS ~]$ date
Tue 03 Mar 2020 12:45:02 PM CET
[nojmy@Nojmy-XPS ~]$
```

I use the BTRFS file system and I have `/tmp` in RAM (mounted as `tmpfs`). Maybe that is somehow related to this issue?

**Comment by andyholmes** [comment]
This should be "fixed" in ee5e5ea, but this really shouldn't happen anyways. There should be no issue with overwriting files with the functions we use for installing files, so my guess is something funky is happening with either the directory permissions or the symlinking of those directories.

In any case, errors are now being caught during `init()` so only truly fatal errors will prevent the extension from starting.

**Comment by kogli** [comment]
@andyholmes Thank you for such a quick response. :)
